### PR TITLE
Add OperationWithObject

### PR DESCRIPTION
**Pull request by alrom**

### CODE REVIEW

File: OutOfSchool/OutOfSchool.DataAccess/Enums/OperationWithObjectEntityType.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace OutOfSchool.Services.Enums;
+
+[JsonConverter(typeof(StringEnumConverter))]
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.DataAccess/Enums/OperationWithObjectEntityType.cs:6]: as this is part of internal job management and not exposed to outside world do we need to have string conversion here?:)

File: OutOfSchool/OutOfSchool.DataAccess/Enums/OperationWithObjectOperationType.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace OutOfSchool.Services.Enums;
+
+[JsonConverter(typeof(StringEnumConverter))]
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.DataAccess/Enums/OperationWithObjectOperationType.cs:6]: same as above

File: OutOfSchool/OutOfSchool.WebApi/Models/OperationWithObjectDto.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+﻿using System.ComponentModel.DataAnnotations;
+using OutOfSchool.Services.Enums;
+
+namespace OutOfSchool.WebApi.Models;
+
+public class OperationWithObjectDto
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.WebApi/Models/OperationWithObjectDto.cs:6]: does it really need to have all fields mapped from data access at the moment? i've only seen the usage of 2-3 so far

File: OutOfSchool/OutOfSchool.WebApi/Services/IOperationWithObjectService.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    /// <param name="filter">Filter.</param>
+    /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+    Task<IEnumerable<OperationWithObjectDto>> GetAll(OperationWithObjectFilter filter);
+
+    /// <summary>
+    /// Check if entity exists in DB.
+    /// </summary>
+    /// <param name="filter">Filter.</param>
+    /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+    Task<bool> IsExists(OperationWithObjectFilter filter);
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.WebApi/Services/IOperationWithObjectService.cs:50]: `IsExists` does not sound good :) can change to just `Exists` or some other word with `Is` prefix :)

File: OutOfSchool/OutOfSchool.WebApi/Services/AverageRatings/AverageRatingService.cs [hunk 5]
```diff
@@ -134,14 +141,25 @@ private async Task<IEnumerable<RatingDto>> GetNewRatings(DateTimeOffset lastLaun
...
+        workshopsIds = workshopsIds.Union(workshopsIdsFromOperations);
+
         var providersIds = GetUniqueProviderIdsByWorkshopIdsAsync(workshopsIds);
 
         Dictionary<Guid, Tuple<float, int>> averageRatings = new Dictionary<Guid, Tuple<float, int>>();
 
         averageRatings.AddRange(CalculateWorkshopsRatingsByEntityIdsAsync(workshopsIds));
         averageRatings.AddRange(CalculateProvidersRatingsByProvidersIdsAsync(providersIds));
 
+        foreach (var operation in operations)
```
Comment (DmyMi) [inline OutOfSchool/OutOfSchool.WebApi/Services/AverageRatings/AverageRatingService.cs:158]: maybe run this in transaction with what ever logic it runs in?

or it does not matter if we fail to delete a rating operation and just try it later some time?

**Review by DmyMi — APPROVED** [review]

File: OutOfSchool/OutOfSchool.DataAccess/Enums/OperationWithObjectEntityType.cs (hunk 1 above)
Comment (alrom) [reply OutOfSchool/OutOfSchool.DataAccess/Enums/OperationWithObjectEntityType.cs:6]: I thought in the future we'll make a controller for ours front's tasks )))) But, OK.

File: OutOfSchool/OutOfSchool.WebApi/Services/AverageRatings/AverageRatingService.cs (hunk 5 above)
Comment (alrom) [reply OutOfSchool/OutOfSchool.WebApi/Services/AverageRatings/AverageRatingService.cs:158]: The last one ))))

**Pull request merged**